lib/storage/metadata/oplog/BucketdOplogInterface.js

```diff
@@ -0,0 +1,251 @@
+/*
+ * Main interface for bucketd oplog management
+ */
+const async = require('async');
+const BucketClient = require('bucketclient').RESTClient;
+const { jsutil } = require('arsenal');
+const LogConsumer = require('arsenal').storage.metadata.bucketclient.LogConsumer;
+const { isMasterKey } = require('arsenal/lib/versioning/Version');
+const OplogInterface = require('./OplogInterface');
+
+class BucketdOplogInterface extends OplogInterface {
+
+    constructor(params) {
+        super(params);
+        this.stopAt = -1;
+        this.backendRetryTimes = 3;
+        this.backendRetryInterval = 300;
+        this.bucketdOplogQuerySize = 20;
+        let bkBootstrap = ['localhost:9000'];
+        if (params && params.bootstrap !== undefined) {
+            bkBootstrap = params.bootstrap;
+        }
+        if (params && params.stopAt !== undefined) {
+            this.stopAt = params.stopAt;
+        }
+        this.bkClient = new BucketClient(bkBootstrap);
+    }
+
+    start(bucketName, cb) {
```

Comment:
Instead of having a `bucketName` there could be a more generic `filter` with a list of buckets or a raftId.